Grbl_Esp32/src/NutsBolts.h

```diff
@@ -20,28 +20,28 @@
   along with Grbl.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-#include "Config.h"
+// #include "Config.h"
 
 // #define false 0
 // #define true 1
 
-#define SOME_LARGE_VALUE 1.0E+38
+const double SOME_LARGE_VALUE = 1.0E+38;
 
 // Axis array index values. Must start with 0 and be continuous.
 // Note: You set the number of axes used by changing N_AXIS.
 // Be sure to define pins or servos in the machine definition file.
-#define X_AXIS 0  // Axis indexing value.
-#define Y_AXIS 1
-#define Z_AXIS 2
+const int X_AXIS = 0;  // Axis indexing value.
+const int Y_AXIS = 1;
+const int Z_AXIS = 2;
 #define A_AXIS 3
```

Comment:
I think that A_AXIS .. C_AXIS can be const int too.  The numbers are not supposed to change.  I tried the change and compiled.

Reply:
Yes and no. The issue is `#if (N_AXIS > A_AXIS)`. Because A_AXIS is not defined (it's a const after all), it will produce different code.

The way to solve it is by using: `if constexpr (N_AXIS > A_AXIS)` or just `if (N_AXIS > A_AXIS)`. It's not trivial, but trivial enough...

Reply:
if constexpr  doesn't work with the compiler we use; I guess it is a C++17 feature.

if doesn't work either because the arrays are too small for the index value in some cases.

So I suppose that we are stuck with the #defines for now.  The problem will go away when the loops are re-rolled and the arrays are dimensioned by MAX_N_AXES.